binsync/core/client.py

```diff
@@ -106,6 +108,7 @@ def __init__(
         self.master_user = master_user
         self.repo_root = repo_root
         self.binary_hash = binary_hash
+        self.binary_path = binary_path
```

Comment:
My biggest concern here is why are we adding `binary_path` to the `Client` class? Client is a class that deals specifically with Git projects and the general database for BinSync. It should not really have any awareness of the binary. 

I realize the last implementation has the `_load_or_update_config` inside the `Client`, but we should very likely move that out and into Controller, since the controller has both awareness of the Client and of the binary that's being worked on.